core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -368,9 +369,11 @@
 
   private static Properties loadProperties() {
     Properties saffronProperties = new Properties();
+    ClassLoader classLoader = MoreObjects.firstNonNull(
+        Thread.currentThread().getContextClassLoader(),
+        CalciteSystemProperty.class.getClassLoader());
     // Read properties from the file "saffron.properties", if it exists in classpath
```

Comment:
Since context ClassLoader is customizable, what if it's not null but saffron.properties can only be loaded from `CalciteSystemProperty.class.getClassLoader()` ?

Reply:
I am not too concerned about this, since Log4J uses the same way to load configuration file.